browser-tests/utils/testcafe.utils.ts

```diff
@@ -0,0 +1,63 @@
+import { screen, within } from '@testing-library/testcafe';
+import pretty from 'pretty';
+import TestController, { ClientFunction } from 'testcafe';
+
+export const screenContext = (t: TestController): typeof screen =>
+  injectCallback(screen, (key, params) => {
+    const [matcher, options] = params;
+    t.ctx[key?.startsWith('get') ? 'within' : 'latestSearch'] = {
+      matcher,
+      key,
+      ...(options && { options }),
+    };
+  });
+
+export const withinContext = (t: TestController): typeof within => (selector) =>
+  injectCallback(within(selector), (key, params) => {
+    t.ctx.latestSearch = {
+      ...(t.ctx.within && { within: t.ctx.within }),
+      key,
+      params: params?.length === 1 ? params[0] : params,
+    };
+  });
+
+const injectCallback = <Key extends string, Obj extends Record<Key, unknown>>(
```

Comment:
`<Key extends string, Obj extends Record<Key, unknown>>`

Not sure why that works but:

`<Obj extends Record<Key, unknown>>` doesn't

at least this doesn't give more type safety

Reply:
fixed